checkbox-support/checkbox_support/scripts/run_watcher.py

```diff
@@ -93,124 +90,171 @@ def run(self):
+            - The name of partition. e.g. sda1, nvme1n1p1 etc...
         """
-        logger.error(
-            "no {} storage {} was reported in systemd journal".format(
-                self.args.storage_type, self.args.testcase
+        plainbox_session_share = os.environ.get("PLAINBOX_SESSION_SHARE")
+        if not plainbox_session_share:
+            logger.error("no env var PLAINBOX_SESSION_SHARE")
+            sys.exit(1)
+
+        # TODO: Should name the file by the value of storage_type variable as
+        #       prefix. e.g. thunderbolt_insert_info, mediacard_insert_info.
+        #       Since usb_insert_info is used by usb_read_write script, we
+        #       should refactor usb_read_write script to adopt different files
+        file_name = "usb_insert_info"
+
+        # backup the storage partition info
+        if reserve and mounted_partition:
+            logger.info(
+                "cache file {} is at: {}".format(
+                    file_name, plainbox_session_share
+                )
             )
-        )
-        sys.exit(1)
+            file_to_share = open(
+                os.path.join(plainbox_session_share, file_name), "w"
+            )
+            file_to_share.write(mounted_partition + "\n")
+            file_to_share.close()
+
+        # remove the back info
+        if not reserve:
+            file_to_share = os.path.join(plainbox_session_share, file_name)
+            with contextlib.suppress(FileNotFoundError):
+                os.remove(file_to_share)
 
 
-class USBStorage(StorageInterface):
+class USBStorage(StorageWatcher):
     """
-    USBStorage hanldes the insertion and removal of usb2, usb3 and mediacard.
+    USBStorage handles the insertion and removal of usb2, usb3 and mediacard.
     """
 
-    MOUNTED_PARTITION = None
-    FLAG_DETECTION = {
-        "device": {
-            "new high-speed USB device number": False,
-            "new SuperSpeed USB device number": False,
-            "new SuperSpeed Gen 1 USB device number": False,
-        },
-        "driver": {"using ehci_hcd": False, "using xhci_hcd": False},
-        "insertion": {"USB Mass Storage device detected": False},
-        "removal": {"USB disconnect, device number": False},
-    }
+    class Device(Enum):
+        HIGH_SPEED_USB = "new high-speed USB device"
+        SUPER_SPEED_USB = "new SuperSpeed USB device"
+        SUPER_SPEED_GEN1_USB = "new SuperSpeed Gen 1 USB device"
+
+    class Driver(Enum):
+        USING_EHCI_HCD = "using ehci_hcd"
+        USING_XHCI_HCD = "using xhci_hcd"
+
+    class Action(Enum):
+        INSERTION = "USB Mass Storage device detected"
+        REMOVAL = "USB disconnect, device"
 
     def __init__(self, args):
         self.args = args
+        self.mounted_partition = None
+        self.device = None
+        self.number = None
+        self.driver = None
+        self.action = None
 
-    def do_callback(self, line_str):
+    def callback(self, line_str):
```

Comment:
Please add a docstring that explains how this works, this basically tries to get the detection infos, tries to get the partition info and then tries to see if it got all the info needed to run the test, and if it did, it does. Find some form that is not confused like I'm being here, and write it to a docstring